npc/re/instances/InfiniteSpace.txt

```diff
@@ -0,0 +1,834 @@
+
+						case 3:
+							.@xs = 1;
+							.@ys = -1;
+							break;
+
+						case 4:
+							.@xs = -1;
+							.@ys = -1;
+							break;
+					}
+					makeitem2 .@equip_id[.@i],1,.@map$,.@x + .@xs,.@y + .@ys,0,0,0,0,0,0,0;
+				}
+			}
+		}
+	}
+	end;
+}
+
+1@infi,112,369,3	duplicate(#w10_t)	#w20_t	4_TREASURE_BOX
+1@infi,194,380,3	duplicate(#w10_t)	#w30_t	4_TREASURE_BOX
+1@infi,280,380,3	duplicate(#w10_t)	#w40_t	4_TREASURE_BOX
+1@infi,366,380,3	duplicate(#w10_t)	#w50_t	4_TREASURE_BOX
+// Aegis spawns hard mode specific treasure npcs
+//1@infi,112,369,3	duplicate(#w10h_t)	#w20h_t	4_TREASURE_BOX
+//1@infi,194,380,3	duplicate(#w10h_t)	#w30h_t	4_TREASURE_BOX
+//1@infi,280,380,3	duplicate(#w10h_t)	#w40h_t	4_TREASURE_BOX
+//1@infi,366,380,3	duplicate(#w10h_t)	#w50h_t	4_TREASURE_BOX
+
+1@infi,30,31,0	script	#w1	WARPNPC,2,2,{
+	end;
+
+OnTouch:
+	.@id = atoi(replacestr(strnpcinfo(2),"w",""));
+	.@map$ = getd("'map$");
+	switch (.@id) {
+		case 1: .@x = 30; .@y = 41; break;
+		case 2: .@x = 30; .@y = 73; break;
+		case 3: .@x = 30; .@y = 105; break;
+		case 4: .@x = 30; .@y = 137; break;
+		case 5: .@x = 30; .@y = 220; break;
+		case 6: .@x = 30; .@y = 253; break;
+		case 7: .@x = 30; .@y = 285; break;
+		case 8: .@x = 30; .@y = 317; break;
+		case 9: .@x = 30; .@y = 349; break;
+		case 10: .@x = 112; .@y = 10; break;
+		case 11: .@x = 112; .@y = 41; break;
+		case 12: .@x = 112; .@y = 73; break;
+		case 13: .@x = 112; .@y = 105; break;
+		case 14: .@x = 112; .@y = 137; break;
+		case 15: .@x = 112; .@y = 220; break;
+		case 16: .@x = 112; .@y = 253; break;
+		case 17: .@x = 112; .@y = 285; break;
+		case 18: .@x = 112; .@y = 317; break;
+		case 19: .@x = 112; .@y = 349; break;
+		case 20: .@x = 198; .@y = 10; break;
+		case 21: .@x = 198; .@y = 41; break;
+		case 22: .@x = 198; .@y = 73; break;
+		case 23: .@x = 198; .@y = 105; break;
+		case 24: .@x = 198; .@y = 137; break;
+		case 25: .@x = 198; .@y = 220; break;
+		case 26: .@x = 198; .@y = 245; break;
+		case 27: .@x = 198; .@y = 277; break;
+		case 28: .@x = 198; .@y = 309; break;
+		case 29: .@x = 198; .@y = 341; break;
+		case 30: .@x = 280; .@y = 10; break;
+		case 31: .@x = 280; .@y = 41; break;
+		case 32: .@x = 280; .@y = 73; break;
+		case 33: .@x = 280; .@y = 105; break;
+		case 34: .@x = 280; .@y = 137; break;
+		case 35: .@x = 280; .@y = 220; break;
+		case 36: .@x = 280; .@y = 245; break;
+		case 37: .@x = 280; .@y = 277; break;
+		case 38: .@x = 280; .@y = 309; break;
+		case 39: .@x = 280; .@y = 341; break;
+		case 40: .@x = 362; .@y = 10; break;
+		case 41: .@x = 362; .@y = 41; break;
+		case 42: .@x = 362; .@y = 73; break;
+		case 43: .@x = 362; .@y = 105; break;
+		case 44: .@x = 362; .@y = 137; break;
+		case 45: .@x = 362; .@y = 220; break;
+		case 46: .@x = 362; .@y = 245; break;
+		case 47: .@x = 362; .@y = 277; break;
+		case 48: .@x = 362; .@y = 309; break;
+		case 49: .@x = 362; .@y = 341; break;
+		default: end;
+	}
+	warp .@map$,.@x,.@y;
+	if (is_party_leader())
```

Comment:
i would like an extra check (maybe with instance variable?). imagine the leader can back to the previous room this event would be triggered again
prevention is better

Reply:
There's currently no way to go back.
We could add an insanity check as you suggested.